collect_app/src/main/java/org/odk/collect/android/database/DatabaseConnection.kt

```diff
@@ -26,29 +27,28 @@ open class DatabaseConnection(
     val readableDatabase: SQLiteDatabase
         get() = dbHelper.readableDatabase
 
-    private val dbHelper: SQLiteOpenHelper by lazy {
-        getOpenHelper(path + name) {
-            DatabaseMigratorSQLiteOpenHelper(
-                AltDatabasePathContext(path, context),
-                name,
-                null,
-                databaseVersion,
-                migrator
-            )
+    private val dbHelper: SQLiteOpenHelper
+        get() {
+            val databasePath = path + File.separator + name
+            if (openHelpers.containsKey(databasePath) && !File(databasePath).exists()) {
+                openHelpers.remove(databasePath)
```

Comment:
Do you think we need to call `close()` on this invalid "open" helper now? It feels like we must be leaking something otherwise.

Reply:
Yeah I think you are right.